javaagent-tooling/src/main/java/io/opentelemetry/javaagent/tooling/config/ConfigPropertiesBridge.java

```diff
@@ -111,9 +113,26 @@ public Map<String, String> getMap(String name, Map<String, String> defaultValue)
 
   @Nullable
   @Override
-  public DeclarativeConfigProperties getDeclarativeConfig(String instrumentationName) {
-    return configProvider != null
-        ? InstrumentationConfigUtil.javaInstrumentationConfig(configProvider, instrumentationName)
-        : null;
+  public DeclarativeConfigProperties getDeclarativeConfig(String node) {
+    if (configProvider == null) {
+      // declarative config is not used
+      return null;
+    }
+
+    DeclarativeConfigProperties config =
+        InstrumentationConfigUtil.javaInstrumentationConfig(configProvider, node);
+    if (config == null) {
+      // there is no declarative config for this node
+      // this needs to be a different value than null to avoid confusion with
+      // the case when declarative config is not supported at all
```

Comment:
it could be confusing to differentiate between null and empty, not sure if there's another option though...

Reply:
yes, we can add `isDeclarativeConfigUsed`!

Reply:
This is the resulting usage: 

```java
    List<TypeInstrumentation> list =
        config.isDeclarative()
            ? MethodsConfig.parseDeclarativeConfig(config.getDeclarativeConfig("methods"))
            : parseConfigProperties();
```